cmd/snap/main.go

```diff
@@ -473,5 +473,7 @@ fixed-width fonts, so it can be hard to tell.
 		fmt.Fprintln(Stderr, msg)
 	}
 
+	checkWarnings(cli.WarningsSummary())
```

Comment:
Would it make sense to simply pass `cli` in here? Or even to make a `client.WarningSummary(w io.Writer)` or similar?